cfme/tests/infrastructure/test_tenant_quota.py

```diff
@@ -135,26 +137,38 @@ def test_tenant_quota_enforce_via_lifecycle(appliance, provider, setup_provider,
 # sequence is important here
 # indirect is the list where we define which fixtures are to be passed values indirectly.
 @pytest.mark.parametrize(
-    ['set_roottenant_quota', 'custom_prov_data', 'extra_msg'],
+    ['set_roottenant_quota', 'custom_prov_data', 'extra_msg', 'context'],
     [
-        [('cpu', '2'), {'hardware': {'num_sockets': '8'}}, ''],
-        [('storage', '0.01'), {}, ''],
-        [('memory', '2'), {'hardware': {'memory': '4096'}}, ''],
-        [('vm', '1'), {'catalog': {'num_vms': '4'}}, '###']
+        [('cpu', '2'), {'hardware': {'num_sockets': '8'}}, '', ViaUI],
+        [('cpu', '2'), {'hardware': {'num_sockets': '8'}}, '', ViaSSUI],
+        [('storage', '0.01'), {}, '', ViaUI],
+        [('storage', '0.01'), {}, '', ViaSSUI],
+        [('memory', '2'), {'hardware': {'memory': '4096'}}, '', ViaUI],
+        [('memory', '2'), {'hardware': {'memory': '4096'}}, '', ViaSSUI],
+        [('vm', '1'), {'catalog': {'num_vms': '4'}}, '###', ViaUI],
+        [('vm', '1'), {'catalog': {'num_vms': '4'}}, '###', ViaSSUI]
     ],
     indirect=['set_roottenant_quota'],
-    ids=['max_cpu', 'max_storage', 'max_memory', 'max_vms']
+    ids=['max_cpu-ViaUI', 'max_cpu-ViaSSUI', 'max_storage-ViaUI', 'max_storage-ViaSSUI',
+         'max_storage-ViaUI', 'max_storage-ViaSSUI', 'max_vms-ViaUI', 'max_vms-ViaSSUI']
 )
 def test_tenant_quota_enforce_via_service(request, appliance, provider, setup_provider,
-                                          set_roottenant_quota, extra_msg, custom_prov_data,
-                                          prov_data, template_name, catalog_item):
+                                          set_roottenant_quota, extra_msg, context,
+                                          custom_prov_data, prov_data, template_name, catalog_item):
     """Test Tenant Quota in UI"""
     catalog_item.provisioning_data.update(custom_prov_data)
     catalog_item.provisioning_data['catalog']['vm_name'] = catalog_item.vm_name
     catalog_item.create()
     service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog,
                                        catalog_item.name)
-    service_catalogs.order()
+    if context is ViaUI:
```

Comment:
**REQUIRED:** Sooooo :) looking at this test I see a few things. You are trying to do contextual parametrization, which is great, however there are a few things here.

It is definitely preferable, in fact almost essential, that the whole of the test case is inside the context manager. This is because I think eventually we will move the context manager outside into the pytest wrapper so that it won't even have to be specified. However it does mean that everything inside this test will be executed inside the contextual manager which would break lines 159-163.

These lines should actually be a fixture because they are not related to the test itself. It's possible we could actually use a preexisting fixture for this.

The finalizer should also be a teardown fixture only. This way we can move the entire body into the context manager.